### PR TITLE
chore(insights): migrate insights v1.x with perso and latest endpoint

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This PR upgrades the insights client and update the implem to provide the right parameters (including indexName, queryID and objectID).

Open questions:
- what event name should we pick for the user? how to make them configurable
- shall we add any tracking to filters, which will truely enable perso?
- how to pass the userToken (can we use the magento customer ID?)
- How do we hook this up to the checkout page?

**Result**

Click and conversion events are sent in theses 3 senarios:
  1. User searched in the autocomplete, clicks on a product, then adds it to cart

  2. User is on the search page, clicks on add to cart:

  3. User is on the search page, clicks on a product, then adds it to cart

### CODE REVIEW

**Comment by tkrugg** [comment]
hey @betty, AFAIKT this isn't failing because if this PR in particular, or is it?

File: view/frontend/web/click_conversion_analytics.js [hunk 1]
```diff
@@ -1,90 +1,100 @@
...
-				trackConversion(objectId);
-			});
-		}
+		// FIXME: reenable this code later
+		// if (algoliaConfig.ccAnalytics.conversionAnalyticsMode === 'place_order'
+		// 	&& algoliaConfig.ccAnalytics.orderedProductIds.length > 0) {
+		// 	$.each(algoliaConfig.ccAnalytics.orderedProductIds, function (i, objectId) {
+		// 		trackConversion(objectId);
+		// 	});
+		// }
```
Comment (tkrugg) [inline view/frontend/web/click_conversion_analytics.js]: This functionality needs backend logic to work.

File: view/frontend/web/click_conversion_analytics.js [hunk 2]
```diff
@@ -1,90 +1,100 @@
...
-	
-	function trackConversion(objectID) {
-		algoliaAnalytics.conversion({
-			objectID: objectID.toString()
-		});
+
+	function trackConversion(objectID, queryID, indexName) {
+		var conversionData = {
+			eventName: 'Added to cart',
+			objectIDs: [objectID + ''],
```
Comment (bsuravech) [inline view/frontend/web/click_conversion_analytics.js]: @tkrugg What's the need for `+ ''` here?

**Comment by tkrugg** [comment]
closed in favor of #895

**Pull request closed without merging**